### PR TITLE
Adds pre-commit checks to the repo -- SGPROJ-1570

### PR DESCRIPTION
**Pull request by DJKaswa**
## Change description

> Please review this and feel free to fix anything that will block it from passing on Github; otherwise it should be good to go as it passes all checks locally.

> Once you review and approve it, you can merge it to main (don’t forget to delete the branch). At that point, it’ll be ready to be moved to tf-auto.

## Type of change
- [ ] Bug fix (fixes an issue)
- [x] New feature (adds functionality)

## Related issues

> Fix [#1]() 

### Development

- [x] Lint rules pass locally
- [ ] Application changes have been tested thoroughly
- [x] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review 

- [x] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [x] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [x] Pull request linked to task tracker where applicable

### CODE REVIEW

File: config.tf [hunk 1]
```diff
@@ -160,11 +161,11 @@ EOF
 }
 
 #Cloudwatch Log Group for Function
+#tfsec:ignore:aws-cloudwatch-log-group-customer-key -- Ignores warning on CMK encryption of CloudWatch Log Groups
```
Comment (JoseMotto-sg) [inline config.tf]: Please create a KMS key for encrypting the resource
```suggestion
```

File: examples/example1.tfnot [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  redis_targets = [
+    "redis-cluster-001.yhglom.0001.use1.cache.amazonaws.com",
+    "redis-cluster-1-002.yhglom.0001.use1.cache.amazonaws.com",
+    "redis-cluster-1-003.yhglom.0001.use1.cache.amazonaws.com"
+  ]
+  name                = "${var.name_prefix}-redis-slowlog-to-cw-logs${local.name_suffix}"
+  memory_size         = 128
+  input_tags          = merge(local.common_tags, {})
+  target_log_group    = "/aws/elasticache/redis/slowlogs" # This will be automatically created when ran if it doesn't already exist
+  subnet_ids          = ["subnet-fjf4b1b","subnet-fgtyuc1e"]
```
Comment (JoseMotto-sg) [inline examples/example1.tfnot]: Use dummy names
```suggestion
  subnet_ids          = ["subnet-00000","subnet-00000"]
```

File: examples/example1.tfnot [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  security_group_ids  = ["sg-fjusg9d32"] # Needs to be able to reach all redis targets and internet via NAT GW
```
Comment (JoseMotto-sg) [inline examples/example1.tfnot]: Use dummy names
```suggestion
  security_group_ids  = ["sg-00000"] # Needs to be able to reach all redis targets and internet via NAT GW
```

File: examples/example2.tfnot [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{
+    "id": 609648,                 <<< unique ID of Cmd
+    "timestamp": 1586322428,      <<< Timestamp in unix time
+    "duration": 37459,            <<< Time in microseconds (so, 37.459 ms)
+    "cmdArray": [                 <<< Cmd Array
+        "GET",
+        ":1:key:your-key"
+    ],
+    "client": {
+        "ip": "172.3.9.44",       <<< Client IP
```
Comment (JoseMotto-sg) [inline examples/example2.tfnot]: Use dummy IPs
```suggestion
        "ip": "172.1.1.1",       <<< Client IP
```

**Review by JoseMotto-sg — CHANGES_REQUESTED** [review]

**Comment by DJKaswa** [comment]
@JoseMotto-sg please review again.

**Comment by JoseMotto-sg** [comment]
Great Work!

**Review by JoseMotto-sg — APPROVED** [review]
LGTM

**Pull request merged**